sympy/core/function.py

```diff
@@ -1257,8 +1257,6 @@ def __new__(cls, expr, *variables, **kwargs):
                     old_v = v
                     v = Dummy('xi_%i' % len(subs), dummy_index=hash(v))
                     expr = expr.xreplace({old_v: v})
-                    if not expr.has(v):
```

Comment:
It seems that some sort of trickery is going on with vector objects. If you run this docstring test and watch expr you will see that it has a `u1` printed but the replace doesn't `u1(t)` with the dummy...so it looks like the answer should be 0 instead of `A.x`:

```python
>>> from sympy.physics.vector import ReferenceFrame, dynamicsymbols
>>> N = ReferenceFrame('N')
>>> A = ReferenceFrame('A')
>>> u1, u2 = dynamicsymbols('u1, u2')
>>> A.set_ang_vel(N, u1 * A.x + u2 * N.y)
>>> A.partial_velocity(N, u1)
A.x
```

I suspect that a "dynamicsymbol" has an explicit time dependence that its `_eval_derivative` knows about, 

This whole routine seems more complicated than it should be in keeping track of how different objects should be handled...